commit ad87cf95df212d2535cfd03c76df58684c6d6e03
Author: Dave Stevenson <dave.stevenson@raspberrypi.org>
Date:   Tue Apr 10 12:39:06 2018 +0100

    net: lan78xx: Add support for VLAN filtering.
    
    HW_VLAN_CTAG_FILTER was partially implemented, but not fully to Linux.
    Complete the implementation of this.
    
    See #2458.
    
    Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.org>

diff --git a/drivers/net/usb/lan78xx.c b/drivers/net/usb/lan78xx.c
index 93adaaa6de4e..30703df0f9d3 100644
--- a/drivers/net/usb/lan78xx.c
+++ b/drivers/net/usb/lan78xx.c
@@ -2295,7 +2295,7 @@ static int lan78xx_set_features(struct net_device *netdev,
 		pdata->rfe_ctl &= ~(RFE_CTL_ICMP_COE_ | RFE_CTL_IGMP_COE_);
 	}
 
-	if (features & NETIF_F_HW_VLAN_CTAG_RX)
+	if (features & NETIF_F_HW_VLAN_CTAG_FILTER)
 		pdata->rfe_ctl |= RFE_CTL_VLAN_FILTER_;
 	else
 		pdata->rfe_ctl &= ~RFE_CTL_VLAN_FILTER_;
@@ -2909,6 +2909,9 @@ static int lan78xx_bind(struct lan78xx_net *dev, struct usb_interface *intf)
 	if (DEFAULT_TSO_CSUM_ENABLE)
 		dev->net->features |= NETIF_F_TSO | NETIF_F_TSO6 | NETIF_F_SG;
 
+	if (DEFAULT_VLAN_FILTER_ENABLE)
+		dev->net->features |= NETIF_F_HW_VLAN_CTAG_FILTER;
+
 	dev->net->hw_features = dev->net->features;
 
 	ret = lan78xx_setup_irq_domain(dev);
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1237_ad87cf95df212d2535cfd03c76df58684c6d6e03.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1237_ad87cf95df212d2535cfd03c76df58684c6d6e03.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
